### ISSUE TITLE
on_input called once per frame while key is held

### ISSUE THREAD

**Issue by j-h-a** [defold/defold#8895]
**Describe the bug (REQUIRED)**

`on_input` is called every frame while a bound key is held, but it should only be called when the key is pressed, repeated, or released.

**To Reproduce (REQUIRED)**

1. Create an empty project
2. add one key to the input binding and remove the mouse binding
3. add a single game object
4. create and add a script to the game object
5. paste this code into the script
```
function init(self)
	msg.post(".", "acquire_input_focus")
end

function on_input(self, action_id, action)
	print(action_id, "pressed:", action.pressed, "released:", action.released)
end
```
6. Run the app
7. press and hold the bound key
8. Look at the console output to see bug being demonstrated

**Expected behavior (REQUIRED)**

You should see just one line of output in the console:
`DEBUG:SCRIPT: hash: [my_action] pressed: true released: false`

**Actual behavior**

You see the expected line in the console, then another line getting repeated every frame:
`DEBUG:SCRIPT: hash: [my_action] pressed: true released: false`
`[n++] DEBUG:SCRIPT: hash: [my_action] pressed: false released: false`

**Defold version (REQUIRED):**
  - 1.8.0

**Platforms (REQUIRED):**
 - macOS Sonoma 14.4.1
 - Mac

**Minimal repro case project (OPTIONAL):**
[input_bug.zip](https://github.com/defold/defold/files/15243772/input_bug.zip)

**Comment by britzl** [comment]
Ah, no, the `pressed` and `released` states happen only once on the frame when the input was first pressed or released. It will not repeat. Note that there is also a `repeated` state which will be triggered at the configured [Repeat Interval](https://defold.com/manuals/project-settings/#repeat-interval)

https://defold.com/ref/go/#on_input:self-action_id-action

**Comment by britzl** [comment]
> I think this was closed without understanding the issue/report. Perhaps images will help so that you don't have to run the minimal repro project.

Sorry, I did indeed misunderstand the problem. We'll take a look. This seems like a regression.

**Comment by drajjon** [comment]
Not sure worth opening another issue, but I also noticed (in addition to this false/false/false issue) that "repeated" is set to "true" on the initial keypress. This goes against my expectations - I would expect "repeated" to only be true on repeats, not initial press. The documentation implies the latter but doesn't explicitly state either way.

**Comment by britzl** [comment]
> This seems like a regression.

I'm taking a closer look, but it seems like the behavior hasn't changed and that this is not a regression. It has likely worked like this always.

> Expected result (as per documentation):

Which documentation are you referring to?


> I also noticed (in addition to this false/false/false issue) that "repeated" is set to "true" on the initial keypress

This seems odd though!

**Comment by britzl** [comment]
> I'm taking a closer look, but it seems like the behavior hasn't changed and that this is not a regression. It has likely worked like this always.

It's likely that this was caused by the glfw for mac update made around that time though...
